internal/configprovider/manager.go

```diff
@@ -376,37 +379,51 @@ func (m *Manager) parseStringValue(ctx context.Context, s string) (interface{},
 
 			switch {
 			case s[j+1] == expandPrefixChar:
-				// Escaping the prefix so $$ becomes a single $ without attempting
-				// to treat the string after it as a config source or env var.
-				expandableContent = string(expandPrefixChar)
-				w = 1 // consumed a single char
+				// temporary backward compatibility to support updated $${config_source:value} functionality
+				// in provided configs from 0.37.0 until 0.42.0
+				bwCompatibilityRequired := false
+
+				var expanded, sourceName string
+				var ww int
+				if ddBackwardCompatible && len(s[j+1:]) > 2 {
+					if s[j+2] == '{' {
+						if expanded, ww, sourceName = getBracketedExpandableContent(s, j+2); sourceName != "" {
+							bwCompatibilityRequired = true
+						}
+					} else {
+						if expanded, ww, sourceName = getBareExpandableContent(s, j+2); sourceName != "" {
+							if len(expanded) > (len(sourceName) + 1) {
+								if !strings.ContainsAny(expanded[len(sourceName)+1:], "$:") {
+									bwCompatibilityRequired = true
+								}
+							}
+						}
+					}
```

Comment:
What is the use case we are supporting in this `else` section? Look like all we need is `$${<source>:<name>}` which in handled in under `if s[j+2] == '{'` clause above, no?

Reply:
The bare (`$env:ENVVAR`) case also suffered from double expansion so required `$$`. I think this was under the radar since we didn't use the form in our configs.

0.41.0
```
single-dollar-no-curly-braces: prefix-$env:AN_ENVVAR
- evaluates to ->
single-dollar-no-curly-braces: prefix-:AN_ENVVAR

and

double-dollar-no-curly-braces: prefix-$$env:AN_ENVVAR
- evaluates to ->
double-dollar-no-curly-braces: prefix-an-envvar-value
```

proposed changes w/ compatibility:
```
single-dollar-no-curly-braces: prefix-$env:AN_ENVVAR
- evaluates to ->
single-dollar-no-curly-braces: prefix-an-envvar-value

and

double-dollar-no-curly-braces: prefix-$$env:AN_ENVVAR
- evaluates to ->
double-dollar-no-curly-braces: prefix-an-envvar-value
```

I've updated the tests to include this form

Reply:
Ok, I didn't know this pattern supported :)